admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/ConfigurationAdmin.java

```diff
@@ -170,32 +185,37 @@ public int compare(AdminModule adminModule, AdminModule adminModule2) {
                 return adminModule.getName().compareTo(adminModule2.getName());
             }
         });
+
         List<Map<String, Object>> modules = new ArrayList<Map<String, Object>>();
         HashMap<String, Object> module;
+
         for (AdminModule adminModule : adminModuleList) {
             module = new HashMap<String, Object>();
             module.put("name", adminModule.getName());
             module.put("id", adminModule.getId());
-            if (adminModule.getJSLocation() != null) {
-                module.put("jsLocation", adminModule.getJSLocation().toString());
-            } else {
-                module.put("jsLocation", "");
-            }
-            if (adminModule.getCSSLocation() != null) {
-                module.put("cssLocation", adminModule.getCSSLocation().toString());
-            } else {
-                module.put("cssLocation", "");
-            }
-            if (adminModule.getIframeLocation() != null) {
-                module.put("iframeLocation", adminModule.getIframeLocation().toString());
+
+            String jsLocation = checkURI(adminModule.getJSLocation());
+            String cssLocation = checkURI(adminModule.getCSSLocation());
+            String iframeLocation = checkURI(adminModule.getIframeLocation());
+
+            module.put("jsLocation", jsLocation);
+            module.put("cssLocation", cssLocation);
+            module.put("iframeLocation", iframeLocation);
+
+            if (StringUtils.isEmpty(jsLocation) &&
```

Comment:
Since we're not going to do anything with invalid modules, we should do this check as soon as possible to avoid doing extra work.